### DISCUSSION TITLE
tethering to nodes in a 2D projected map not working...

### DISCUSSION THREAD

**Discussion by keince** [pelicanmapping/osgearth#1889] [Support Q&A] [answered]
I have a mapNode that is using a spherical mercator map profile (projected 2D map). I'm adding some tracknodes to the mapNode and trying to tether the viewpoint to a selected node by the following...

```
Viewpoint vp = manipulator->getViewpoint();
vp.setNode(tracknode);
vp.range()->set(25000.0, Units::METERS);
vp.pitch()->set(-90.0, Units::DEGREES);
manipulator->setViewpoint(vp, 2.0);
```

For some reason, this doesn't tether to the tracknode as I would expect. When I run the same code for nodes in a geocentric map profile, it works as expected.

Is there something broken with tethering to nodes in projected map profiles? Perhaps I'm needing to perform a transform on the node somewhere first?

Any help or feedback would be appreciated...

**Comment by gwaldron** [comment, marked as answer]
The `osgearth_manip` example is the test app for tethered nodes. You can run this with a projected map (like `openstreetmap_flat.earth` or `readymap_flat.earth`) and press 8 or 9 to tether. 

The problem might be with `TrackNode` so perhaps you can modify the example to use one and see if it still works.

**Reply by keince** [reply]
Glenn,

Thanks for the reply and suggestion. Per your recommendation, I altered osgearth_manip and replaced the following...

1. Changed osg::ref_ptr<GeoPositionNode> _geo with osg::ref_ptr<TrackNode> _geo
2. Changed vp.setNode(_label) to vp.setNode(_geo->getGeoTransform());

Now I can tether to trackNodes!

If I change my code above from vp.setNode(tracknode) to vp.setNode(trackNode->getGeoTransform()), everything works. The key was using the getGeoTransform() call and not the trackNode itself when setting the node to tether in the viewPoint.

Thank you for your help!
